core/src/main/java/com/hevelian/identity/core/TenantService.java

```diff
@@ -104,22 +112,28 @@ public Tenant getTenant(String tenantDomain) throws TenantNotFoundByDomainExcept
 
   public BufferedImage getTenantLogo(String tenantDomain)
       throws TenantNotFoundByDomainException, TenantHasNoLogoException {
-    Tenant tenant = getTenant(tenantDomain);
-    byte[] logo = tenant.getLogo();
+    byte[] logo = getTenantLogoBytes(tenantDomain);
     BufferedImage img = null;
     try {
       if (logo != null) {
         img = ImageIO.read(new ByteArrayInputStream(logo));
       }
     } catch (IOException e) {
-      throw new ReadImageException("Error reading image from byte array input stream.", e);
-    }
-    if (img == null) {
-      throw new TenantHasNoLogoException(tenant.getDomain());
+      throw new ImageRetrievalException("Error reading image from byte array input stream.", e);
     }
     return img;
   }
 
+  public byte[] getTenantLogoBytes(String tenantDomain)
```

Comment:
I think that was a bad idea from my side to suggest this method here. You can remove it. You can set tenant logo either by BufferedImage or by a byte array and always get a logo as a buffered image. If we need a getter for byte array in future - we can add it.

Reply:
Fixed